File: app/src/main/java/com/pietrantuono/podcasts/addpodcast/singlepodcast/view/SinglePodcastView.java
package com.pietrantuono.podcasts.addpodcast.singlepodcast.view;

import com.pietrantuono.podcasts.apis.Episode;

import org.jetbrains.annotations.Nullable;

import java.util.List;

public interface SinglePodcastView {

    void enterWithTransition();

    void enterWithoutTransition();

    void showProgress(boolean show);

    void setEpisodes(@Nullable List<Episode> episodes);

    void exitWithSharedTrsnsition();

    void exitWithoutSharedTransition();

    void setSubscribedToPodcast(Boolean isSubscribed);

    void setTitle(@Nullable String collectionName);
}

File: app/src/main/java/com/pietrantuono/podcasts/addpodcast/singlepodcast/view/SinglePodcastView.kt
package com.pietrantuono.podcasts.addpodcast.singlepodcast.view

import models.pojos.Episode

interface SinglePodcastView {

    var title: String?

    fun enterWithTransition()

    fun enterWithoutTransition()

    fun setEpisodes(episodes: List<Episode>)

    fun exitWithSharedTrsnsition()

    fun exitWithoutSharedTransition()

    fun setSubscribedToPodcast(isSubscribed: Boolean?)

    fun onError(string: String?)

    fun onTimeout()

    fun setState(state: State)
}
